src/include/txn_inline.h

```diff
@@ -909,6 +910,16 @@ __wt_upd_alloc(WT_SESSION_IMPL *session, const WT_ITEM *value, u_int modify_type
         (value != NULL &&
           !(modify_type == WT_UPDATE_RESERVE || modify_type == WT_UPDATE_TOMBSTONE)));
 
+    /*
+     * Ensure that the memory allocated is never less than the size reported by sizeof. Otherwise
+     * bit-exact tools like Google MSan may infer the structure is not completely initialized.
+     */
+    if (value != NULL) {
```

Comment:
Did you consider making this `if (value != NULL && value->size != 0)` and getting rid of the `WT_MAX` macro? I think I'd find that a bit easier to reason about.

Reply:
I chose the WT_UPDATE_SIZE_NOVALUE as the term "empty" struck me as potentially confusing.

Reply:
I retained the `WT_MAX` macro but the context is now different.